source/common/upstream/load_balancer_impl.h

```diff
@@ -155,17 +156,31 @@ class LoadBalancerBase : public LoadBalancer {
   // The total count of healthy hosts across all priority levels.
   uint32_t total_healthy_hosts_;
 
+  // Expected override host statues. Every bit in the OverrideHostStatus represent an enum value of
+  // Host::Health. The specific correspondence is shown below:
+  //
+  // * 0b001: Host::Health::Unhealthy
+  // * 0b010: Host::Health::Degraded
+  // * 0b100: Host::Health::Healthy
+  //
+  // If multiple bit fields are set, it is acceptable as long as the status of override host is in
+  // any of these statuses.
+  const std::bitset<32> override_host_status_{};
```

Comment:
FWIW it doesn't need to be 32-bit long, but it probably doesn't matter that much either.

Reply:
Yes. Only 3 bit is used for now. But in the x64 matchine, the storage size of std::bitset<N> (N > 0) is at least 8 bytes. So I guess it's not a big problem. So I think setting N to 32 or 8 makes little difference. 

```
    std::cout << sizeof(std::bitset<8>) << std::endl;  // 8
    std::cout << sizeof(std::bitset<16>) << std::endl;  // 8
    std::cout << sizeof(std::bitset<32>) << std::endl;  // 8
    std::cout << sizeof(std::bitset<64>) << std::endl;  // 8
    std::cout << sizeof(std::bitset<128>) << std::endl; // 16
```